src/partials/dialog.settings.html

```diff
@@ -11,9 +11,47 @@ <h2 translate>settings.SETTINGS</h2>
         </md-toolbar>
         <md-dialog-content>
             <div class="md-dialog-content">
+                <section>
+                    <md-subheader class="md-accent"><span translate>settings.notifications.NOTIFICATIONS</span>
+                    </md-subheader>
+                    <md-list flex>
+                        <md-list-item>
+                            <md-checkbox
+                                    ng-disabled="!ctrl.notificationApiAvailable || ctrl.notificationPermission === false"
+                                    ng-model="ctrl.desktopNotifications"
+                                    ng-change="ctrl.setWantsNotifications(ctrl.desktopNotifications)"
+                                    aria-label="Show desktop notifications">
+                            </md-checkbox>
+                            <p translate>settings.notifications.SHOW_NOTIFICATIONS</p>
+                        </md-list-item>
+                        <md-list-item>
+                            <md-checkbox
+                                    ng-disabled="!ctrl.desktopNotifications"
+                                    ng-model="ctrl.notificationPreview"
+                                    ng-change="ctrl.setWantsPreview(ctrl.notificationPreview)"
+                                    aria-label="Show message preview">
+                            </md-checkbox>
+                            <span translate>settings.notifications.SHOW_PREVIEW</span>
+                        </md-list-item>
+                    </md-list>
 
+                    <div ng-if="!ctrl.notificationApiAvailable">
+                        <span translate>error.NOTIFICATION_API_NOT_AVAILABLE</span>
+                    </div>
```

Comment:
I would actually hide the entire settings options if the API is not available. There's no point in showing options to the user that have no effect. Especially since the first option is visible and checked, but greyed out if the API is not available.